fix(curriculum): correct factual and grammar errors in  lecture-understanding-asynchronous-programming  block (#67569)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/lecture-understanding-asynchronous-programming/6733b072bd8f5b06ccdbd9e2.md

In synchronous programming, threads execute sequentially, one after the other. If a thread is blocked, like waiting for user input, the entire process is blocked until the thread is completed.

JavaScript is a single-threaded programming language that uses an event loop to handle asynchronous operations concurrently without blocking the main execution thread.

> An event loop is a mechanism that continuously monitors the call stack and callback queue, executing queued tasks only when the call stack becomes empty. It enables non-blocking and asynchronous behavior in JavaScript by coordinating the execution of operations such as timers, network requests, and event handlers.
> 
> Some key characteristics of the event loop include continuous execution, task scheduling, queue management, and maintaining application responsiveness while handling multiple operations efficiently.

Reviewer: let's not throw jargon at the camper, let's try to make the explanation friendly instead

this is a quick suggestion, it may need more work

```suggestion
An event loop is a mechanism that continuously monitors the call stack and callback queue, executing queued tasks only when the call stack becomes empty.

In the event loop, the asynchronous tasks are executed later, when there is space for them.
```

In synchronous programming, threads execute sequentially, one after the other. If a thread is blocked, like waiting for user input, the entire process is blocked until the thread is completed.

> JavaScript is a single-threaded programming language that uses an event loop to handle asynchronous operations concurrently without blocking the main execution thread.

Reviewer: can you add something more on the event loop? I don't think it was mentioned previously

Author: I added a bit more context about the event loop and now it briefly gives a lil bit more context on what the event loop
